python/tvm/_ffi/runtime_ctypes.py

```diff
@@ -262,6 +264,35 @@ def max_thread_dimensions(self):
         """
         return json.loads(self._GetDeviceAttr(self.device_type, self.device_id, 8))
 
+    @property
+    def api_version(self):
+        """Returns version number of the SDK used to compile TVM.
+
+        For example, CUDA_VERSION for cuda or VK_HEADER_VERSION for
+        Vulkan.
+
+        Returns
+        -------
+        version : int
+            The version of the SDK
+
+        """
+        return self._GetDeviceAttr(self.device_type, self.device_id, 12)
+
+    @property
+    def driver_version(self):
+        """Returns version number of the driver
+
+        Returns driver vendor's internal version number.
+        (e.g. "450.408.256" for nvidia-driver-450)
+
+        Returns
+        -------
+        version : str
```

Comment:
seems like some implementations return nullptr/None, want to document when and why?

Reply:
Certainly, can do.  It looks like several of the other parameters also return None for some implementations, sometimes for cases where a parameter doesn't make sense (e.g. warp size for cpu targets), or API limitations (e.g. clock speed on vulkan).  I'll see if I can flesh out some of the implementations, and add documentation for each parameter on when they return None.

Reply:
could you be more explicit in the docstring? imo, "Implemented for opencl and vulkan" would read to a Python developer as "raises NotImplementedError when not opencl or vulkan"

Reply:
Good point, I had been thinking of implemented in C++ virtual functions terms.  I'll update the docstrings to instead read "Returns device value for X, Y, and Z devices.  Returns None for any other devices."

Reply:
Also, as long as I am looking at it, it would be good to explicitly state that the RPC devices return the value of the remote device.

Reply:
I think what you added to the description is good, but I still think the "Returns" section is confusing, because it doesn't state "str or None." Someone skimming this would likely miss the body section. Could you add the None part to "Returns," here and elsewhere?

Reply:
see also https://numpydoc.readthedocs.io/en/latest/format.html#sections, which doesn't require you to name return values; this might help with adding multiple types of return values.